src/app/clusters/camera-av-settings-user-level-management-server/camera-av-settings-user-level-management-server.cpp

```diff
@@ -0,0 +1,1079 @@
+        if (newTilt > mTiltMax)
+        {
+            newTilt = mTiltMax;
+        }
+        if (newTilt < mTiltMin)
+        {
+            newTilt = mTiltMin;
+        }
+
+        hasAtLeastOneValue = true;
+    }
+
+    if (zoomDelta.HasValue())
+    {
+        if (!HasFeature(Feature::kMechanicalZoom))
+        {
+            ChipLogError(Zcl, "Mechanical Zoom not supported although zoomDelta value provided");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+            return;
+        }
+        int8_t zoomDeltaValue = zoomDelta.Value();
+        if (zoomDeltaValue > (mZoomMax - 1) || zoomDeltaValue < -(mZoomMax - 1))
+        {
+            ChipLogError(Zcl, "ZoomDelta value received is out of range.");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+            return;
+        }
+
+        newZoom = static_cast<int8_t>(newZoom + zoomDeltaValue);
+        if (newZoom > mZoomMax)
+        {
+            newZoom = static_cast<int8_t>(mZoomMax);
+        }
+        if (newZoom < 1)
+        {
+            newZoom = 1;
+        }
+
+        hasAtLeastOneValue = true;
+    }
+
+    // Was a value received in the command
+    if (!hasAtLeastOneValue)
+    {
+        ChipLogError(Zcl, "MPTZRelativeMove contains no actionable fields");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+        return;
+    }
+
+    // Check with the delegate that we're in a position to change any of the PTZ values
+    //
+    if (!mDelegate->CanChangeMPTZ())
+    {
+        ChipLogDetail(Zcl, "Device not able to process MPTZ relative value change");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Busy);
+        return;
+    }
+
+    // Call the delegate to simply set the newly calculated MPTZ values based on the deltas received
+    //
+    status = mDelegate->MPTZRelativeMove(Optional(static_cast<int16_t>(newPan)), Optional(static_cast<int16_t>(newTilt)),
+                                         Optional(static_cast<uint8_t>(newZoom)));
+
+    if (status != Status::Success)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        return;
+    }
+
+    // Set the local values of pan, tilt, and zoom
+    SetPan(Optional<int16_t>(newPan));
+    SetTilt(Optional<int16_t>(newTilt));
+    SetZoom(Optional<uint8_t>(static_cast<uint8_t>(newZoom)));
+
+    MarkDirty(Attributes::MPTZPosition::Id);
+
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+}
+
+void CameraAvSettingsUserLevelMgmtServer::HandleMPTZMoveToPreset(HandlerContext & ctx,
+                                                                 const Commands::MPTZMoveToPreset::DecodableType & commandData)
+{
+    Status status  = Status::Success;
+    uint8_t preset = commandData.presetID;
+
+    // Verify the provided presetID is within spec limits
+    //
+    if (preset > mMaxPresets - 1)
```

Comment:
Doesn't match the spec in https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/11333/files, so please just make sure that difference is tracked and this is fixed once the spec updates.

Reply:
Fixed in this PR per guidance from Aron.  Same for other comments on presetID checks